### DISCUSSION TITLE
How to build Lc0 without OpenCL support (currently automatically picks it up)

### DISCUSSION THREAD

**Discussion by BuddhiLW** [LeelaChessZero/lc0#1936] [Q&A] [answered]
I'm on Ubuntu 23.10,  CPU: AMD Ryzen 5 5600X (12) @ 3.700GHz, GPU: AMD ATI Radeon RX 470/480/570/570X/580/580X/590

I'm using `nibbler`. When I open choose lc0 that I just built, I get:

```
Plataform version: OpenCL 2.1 AMD-APP (3590.0)
(...)
Error getting device(s): clGetDeviceIDs: -1
Unhandled exception: No Suitable OpenCL device found.
```

The question becomes, how can I build without the `OpenCL` support, specified in the build command? I didn't find anything in the documentation related to specifying this.

**Comment by mooskagh** [comment, marked as answer]
To disable opencl support while building Lc0, pass `-Dopencl=false` flag to the `./build.sh` command (or to `meson setup` if you call it directly).

If you did build Lc0 with the opencl support, you can still use a different backend when running it.
To do that, pass `--backend=<backend>` command line argument or `Backend <backend>` uci parameter (I'm not sure whether it's possible to do in nibbler though), or create a `lc0.config` file either in the same directory as the `lc0` binary, or at `~/.config/lc0/lc0.config`, with the following contents:
```
backend=<backend>
```
To get the list of the available backends, read the description of the `--backend` parameter at `./lc0 --help`. In your case, the most likely backends that may work are `blas` and `eigen`.

`blas` and `eigen` backends will only use CPU though. There is a chance that it's possible to run Lc0 on your GPU, but I'm not sure.  It may be a good idea to ask in the #help channel of our Discord chat at http://lc0.org/chat
